packages/blockchain-wallet-v4-frontend/src/data/goals/sagas.js

```diff
@@ -85,12 +86,21 @@ export default ({ api }) => {
     // Special case to handle bitcoin bip21 link integration
     const decodedPayload = decodeURIComponent(pathname + search)
     const bip21Payload = bip21.decode(decodedPayload)
-    const { address } = bip21Payload
-    const { amount, message } = bip21Payload.options || {}
-    const data = { address, amount, description: message }
-    yield put(actions.goals.saveGoal('payment', data))
-    yield put(actions.router.push('/wallet'))
-    yield put(actions.alerts.displayInfo(C.PLEASE_LOGIN))
+    // check for BitPay payment protocol
+    if (bip21Payload.options && bip21Payload.options.r) {
```

Comment:
prefer Ramda => `if(path(['options', 'r'], bip21Payload)) {`

Reply:
addressed in newest commit